src/modules/producer.js

```diff
@@ -170,12 +171,16 @@ class Producer {
       if (!msg) msg = null;
 
       this._connection.config.transport.info('bmq:producer', `[${queue}] > `, msg);
-
-      return this.checkRpc(queue, parsers.out(msg, options), options);
+      return this.channel.checkQueue(queue).then(() => {
```

Comment:
Huh I think we need to review this one more carefully ( @dial-once/developers  ) @Spring3 I think we tried to do something like this before and it had undesired sideeffects. 
Please read
https://github.com/dial-once/node-bunnymq/pull/49